### PR TITLE
Bump macOs version to macos-11

### PR DESCRIPTION
**Pull request by sima-zhu**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
Resolve https://github.com/Azure/azure-sdk-tools/issues/3715

Tracking issue: https://github.com/actions/virtual-environments/issues/5583
If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/scripts/job-matrix/job-matrix-functions.tests.ps1 [hunk 1]
```diff
@@ -359,15 +359,15 @@ Describe "Platform Matrix Generation" -Tag "generate" {
...
+        $element.parameters.operatingSystem | Should -Be "macos-11"
         $element.parameters.framework | Should -Be "netcoreapp2.1"
         $element.parameters.additionalArguments | Should -Be "--enableFoo"
     }
 
     It "Should initialize a sparse matrix from an N-dimensional matrix" -TestCases @(
         @{ i = 0; name = "windows2019_net461"; operatingSystem = "windows-2019"; framework = "net461"; additionalArguments = ""; }
         @{ i = 1; name = "ubuntu1804_netcoreapp21_withfoo"; operatingSystem = "ubuntu-18.04"; framework = "netcoreapp2.1"; additionalArguments = "--enableFoo"; }
-        @{ i = 2; name = "macOS1015_net461"; operatingSystem = "macOS-10.15"; framework = "net461"; additionalArguments = ""; }
+        @{ i = 2; name = "macOS1015_net461"; operatingSystem = "macos-11"; framework = "net461"; additionalArguments = ""; }
```
Comment (weshaggard) [inline eng/scripts/job-matrix/job-matrix-functions.tests.ps1:370]:
```suggestion
        @{ i = 2; name = "macOS11_net461"; operatingSystem = "macos-11"; framework = "net461"; additionalArguments = ""; }
```

**Review by weshaggard — APPROVED** [review]

**Comment by weshaggard** [comment]
@praveenkuttappan any reason why this PR reported back API changes from APIVIew?

**Review by scbedd — APPROVED** [review]
@sima-zhu I added a commit [here](https://github.com/Azure/azure-sdk-for-python/pull/25410/commits/200ed3ec79011724b272daa892dff7665c597d15) to cover the base platform-matrix. Looks good to me! Thanks for the effort on this!

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "Agent": {
       "ubuntu-20.04": { "OSVmImage": "MMSUbuntu20.04", "Pool": "azsdk-pool-mms-ubuntu-2004-general" },
       "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-win-2019-general" },
-      "macOS-10.15": { "OSVmImage": "macOS-10.15", "Pool": "Azure Pipelines" }
+      "macos-11": { "OSVmImage": "macos-11", "Pool": "Azure Pipelines" }
```
Comment (scbedd) [inline eng/pipelines/templates/stages/platform-matrix.json:11]: I added this @sima-zhu . If you intentionally left it out I apologize!

File: eng/scripts/job-matrix/job-matrix-functions.tests.ps1 [hunk 3]
```diff
@@ -12,7 +12,7 @@ BeforeAll {
         "operatingSystem": [
           "windows-2019",
           "ubuntu-18.04",
-          "macOS-10.15"
+          "macos-11"
```
[the remark this answers is not in the document]
Comment (benbp) [reply eng/scripts/job-matrix/job-matrix-functions.tests.ps1:15]: Wow...not sure how I missed these! https://github.com/Azure/azure-sdk-for-python/pull/25435

File: eng/scripts/job-matrix/job-matrix-functions.tests.ps1 (hunk 1 above)
Comment (benbp) [reply eng/scripts/job-matrix/job-matrix-functions.tests.ps1:370]: See above you can ignore changes to this file as I am removing it.

File: eng/pipelines/templates/stages/platform-matrix.json (hunk 2 above)
Comment (sima-zhu) [reply eng/pipelines/templates/stages/platform-matrix.json:11]: It is definitely needed. Not sure why I missed it.

**Pull request merged**